pkg/handlers/v1/eni.go

```diff
@@ -0,0 +1,115 @@
+}
+
+type privateIP struct {
+	PrivateIPAddress string `json:"PrivateIpAddress"`
+}
+
+type eniTransformer struct{}
+
+func (t eniTransformer) Create(event awsConfigEvent) (Output, error) {
+	output, err := getBaseOutput(event.ConfigurationItem)
+	if err != nil {
+		return Output{}, err
+	}
+
+	var config eniConfiguration
+	if err := json.Unmarshal(event.ConfigurationItem.Configuration, &config); err != nil {
+		return Output{}, err
+	}
+
+	if filter(config) {
+		return output, nil
+	}
+
+	change := extractEniInfo(&config)
+	change.ChangeType = added
+	output.Changes = append(output.Changes, change)
+
+	return output, nil
+}
+
+// Returns true if we should filter this event due to not being requester managed
+func filter(config eniConfiguration) bool {
+	return config.RequesterManaged == false || config.RequesterID != elbManaged
+}
+
+func (t eniTransformer) Update(event awsConfigEvent) (Output, error) {
+	// I don't think requesterManaged ENIs can update in the traditional sense
+	// https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/requester-managed-eni.html
+	// This page implies that they can only exist while attached to whatever requested them
+	// Because this is effectively a no-op no matter what, we don't need to filter for now
+
+	output, err := getBaseOutput(event.ConfigurationItem)
+	if err != nil {
+		return Output{}, err
+	}
+
+	return output, nil
+}
+
+func (t eniTransformer) Delete(event awsConfigEvent) (Output, error) {
+	output, err := getBaseOutput(event.ConfigurationItem)
+	if err != nil {
+		return Output{}, err
+	}
+
+	changeProps := event.ConfigurationItemDiff.ChangedProperties
+	// TODO: Can ARN be null in the ENI config like EC2?
+	// TODO: Get a sample requesterManaged ENI event with tags. I'm not sure if we need tagSet for previousValue
+
+	configDiffRaw, ok := changeProps["Configuration"]
+	if !ok {
+		return Output{}, errors.New("Invalid configuration diff")
+	}
+	var configDiff eniConfigurationDiff
+	if err := json.Unmarshal(configDiffRaw, &configDiff); err != nil {
+		return Output{}, err
+	}
+	if filter(*configDiff.PreviousValue) {
+		return output, nil
+	}
+
+	change := extractEniInfo(configDiff.PreviousValue)
+	change.ChangeType = deleted
+	output.Changes = append(output.Changes, change)
+	return output, nil
+}
+
+func extractEniInfo(config *eniConfiguration) Change {
+	change := Change{}
+
+	// All our sample data only has one privateIp, but it is possible to have multiple
+	privateIPs := []string{}
+	for _, privateIP := range config.PrivateIPAddresses {
+		privateIPs = append(privateIPs, privateIP.PrivateIPAddress)
+	}
+	change.PrivateIPAddresses = append(change.PrivateIPAddresses, privateIPs...)
+
+	// I in general am not a fan of this, but the only other thing I could think of was
+	// regex matching which doesn't seem any better
+	pieces := strings.Split(config.Description, " ")
+	change.RelatedResources = append(change.RelatedResources, pieces[len(pieces)-1])
```

Comment:
question: understanding check for me here, you're splitting on the spaces of a description like `ELB micros-sec-example-ELB-AAAAAA11111` and retrieving only the `micros-sec-example-ELB-AAAAAA1111` portion.

If so, this makes sense to me and is simpler then a regex. The main thing to consider is your counting on the descriptions continuing to be in that format.

Reply:
FWIW I always prefer solving a problem with standard string libraries over a regex when it's appropriate.

Reply:
From what I've seen this description rule will hold true. During the initial research for this back in May(?) Alena found a very helpful AWS knowledge base article about finding the IP address associated with a Load Balancer https://aws.amazon.com/premiumsupport/knowledge-center/elb-find-load-balancer-IP/ The key bit from this article is where they describe how to relate ENIs and LBs by looking at the ENI's description:
`aws ec2 describe-network-interfaces --filters Name=description,Values="ELB elb-name" --query 'NetworkInterfaces[*].PrivateIpAddresses[*].PrivateIpAddress' --output text`

This looks fairly official and I haven't seen any requesterManaged ENI events yet that break this convention.